golang/lint.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: golangci-lint
+spec:
+  inputs:
+    params:
+    - name: package
+      description: base package (and its children) under validation
+    - name: flags
+      description: flags to use for the test command
+      default: --verbose
+    - name: version
+      default: golangci-lint version to use
+      default: "v1.16"
+    - name: GOOS
+      description: "operating system target (default: node operating system)"
+      default:
+    - name: GOARCH
+      description: "architecture target (default: node architecture)"
+      default:
+    resources:
+    - name: source
+      type: git
+      targetPath: src/${inputs.params.package}
+  steps:
+  - name: lint
+    image: golangci/golangci-lint:${inputs.params.version}
```

Comment:
I'm a bit uneasy about recommending this image without knowing much about its provenance. Who is responsible for building and maintaining this image? What prevents them from injecting malicious code?

Reply:
> Who is responsible for building and maintaining this image? What prevents them from injecting malicious code?

@ImJasonH this is true for other tasks like `buildah`, using `quay.io/buildah/stable` (maintained by `containers/buildah` maintainers) or `quay.io/openshift-pipelines/buildah:stable` (maintained by my team). 

This is a good question, and I think it deserves an issue on this repository. Should all the task have the image(s) build recipes with them ? Is `tektoncd/catalog` responsible to build and publish those images ?

Reply:
Yeah I definitely think it's worth documenting at least, as a best practice for reusable Tasks.

In the absence of that, I'm much less worried about referencing an image that's built by some team that maintains the tool. Having a `buildah` image built by Red Hat is fine, having a `gcloud` or `golang` image built by Google is fine, having a `golang`(-adjacent) image built by some Dockerhub user who happens to have the name `golangci` feels...dicey. If it was `golang/lint`, I'd have no real concerns.

Reply:
The `golangci/golangci-lint` image is maintained by the `golangci` team : https://github.com/golangci. In a sense, it's the same as `buildah` built by Red Hat :angel:

Reply:
in a sense this is the same OSS trust we have when adding a golang depdence in the code,

Reply:
I guess that's true. I didn't realize `golanci` was a real company with a real domain. 

This seems fine then. :man_shrugging: